### PR TITLE
Improved SQS Lambda handler partial batch responses - ReportBatchItemFailures

### PR DESCRIPTION
**Pull request by victorskl**
* This feature allows FIFO queue's Lambda trigger to allow partially process
  of messages in a batch and, still keep exception-raised events in the queue.
  Essentially, e.g. total 10 messages; 6 process success and 4 exception raised
  then 6 will clear away from Q; but leave 4 in the Q. Instead of retrying the
  whole batch.

See doc
* https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#services-sqs-batchfailurereporting
* https://repost.aws/knowledge-center/lambda-sqs-report-batch-item-failures

co-co: @reisingerf

### CODE REVIEW

File: data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def test_sqs_handler(self):
         logger.info("Example dragen_tso_ctdna.sqs_handler lambda output:")
         logger.info(json.dumps(results))
 
-        self.assertEqual(len(results), 1)
+        self.assertEqual(len(results), 2)
```
Comment (reisingerf) [inline data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py:215]: Could add a note why we are expecting 2 results, i.e. one for the handler `results` and one for the `batchItemFailures`. No matter what, both should always be present, but may be empty lists.

File: data_processors/pipeline/lambdas/tests/test_dragen_wgs_qc.py [hunk 2]
```diff
@@ -165,7 +165,7 @@ def test_sqs_handler(self):
         logger.info("Example dragen_wgs_qc.sqs_handler lambda output:")
         logger.info(json.dumps(results))
 
-        self.assertEqual(len(results), 1)
+        self.assertEqual(len(results), 2)
```
Comment (reisingerf) [inline data_processors/pipeline/lambdas/tests/test_dragen_wgs_qc.py:168]: Ditto. Add comment why 2

File: data_processors/pipeline/lambdas/tests/test_dragen_wts.py [hunk 3]
```diff
@@ -163,7 +163,7 @@ def test_sqs_handler(self):
         logger.info("Example dragen_wts.sqs_handler lambda output:")
         logger.info(json.dumps(results))
 
-        self.assertEqual(len(results), 1)
+        self.assertEqual(len(results), 2)
```
Comment (reisingerf) [inline data_processors/pipeline/lambdas/tests/test_dragen_wts.py:166]: Ditto. Add comment why 2

**Review by reisingerf — APPROVED** [review]
Perhaps add minor comments, but looks good!

File: data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py (hunk 1 above)
Comment (victorskl) [reply data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py:215]: That's the quiz! j/k 😬 Okie dokie, I will add comment in.

Comment (reisingerf) [reply data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py:215]: Ah, you are just checking if I pay attention... fair enough!

Comment (victorskl) [reply data_processors/pipeline/lambdas/tests/test_dragen_tso_ctdna.py:218]: ⏫ https://github.com/umccr/data-portal-apis/commit/09c4975a447a7590a8ecde135095a371119323b5

**Comment by victorskl** [comment]
[Portal Release - Ocicat: Backend Chores](https://trello.com/c/DuPmROYN/1527-portal-release-ocicat-backend-chores)

**Comment by victorskl** [comment]
Merging...

**Pull request merged**